File: Hashes.py
# Sure enough this is a module

# define a class
class IrreversibleEncryption:

    def __init__(self):
        self.i = 0

    def hashWord(self, msg):
        # encoding eqn { Tn = (n-2)t1 + 2^n } - please use your own eqn
        hashed = 0
        for i in range(len(msg)):
            # get unicode of this character as n
            n = ord(msg[i])
            t1 = i + 1
            # use recurrence series equation to hash
            x = (n - 2) * t1 + 2**n
            if i == 0:
                hashed = x
                continue

            # bitwise rotate left with the modulo of x
            binary = (bin(hashed))[2:] # remove binary designator('0b')
            x %= len(binary)

            slice_1 = list(binary[x:])
            # keep as '1' to preserve hash size
            slice_1[0] = "1"

            slice_2 = binary[0:x]

            hashed = "".join(slice_1) + slice_2
            hashed = int(hashed, 2)

        hashed = (hex(hashed))[2:] # remove hex designator('0x')
        hashed = hashed.upper()

        return hashed

File: Hashes.js
"use strict";

function hashWord(msg) {
    // encoding eqn { Tn = (n-2)t1 + 2^n } - please use your own eqn
    var hash;
    var n;
    var t1;
    var x;
    for (var i = 0; i < msg.length; i++) {
        // get unicode of this character as n
        n = msg.charCodeAt(i);
        t1 = i + 1;
        // use recurrence series equation to hash
        x = bigInt(n).subtract(2).multiply(t1).add(bigInt(2).pow(n));
        if (i === 0) {
            hash = x;
            continue;
        }
        // bitwise rotate left with the modulo of x
        var binary = bigInt(hash).toString(2);
        x = bigInt(x).mod(binary.length);
        
        var slice_1 = binary.slice(x).split('');
        // keep as '1' to preserve hash size
        slice_1[0] = 1;
        slice_1 = slice_1.join('');
        var slice_2 = binary.slice(0, x);
        
        hash = String(slice_1) + String(slice_2);
        hash = bigInt(hash, 2);
    }
    hash = bigInt(hash).toString(16);
    hash = hash.toUpperCase();

    return hash;
}